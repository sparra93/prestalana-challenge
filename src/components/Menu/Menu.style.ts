import styled from 'styled-components';

export const MenuTitle = styled.div`
  text-decoration: none;
  color: #333;
  font-size: 24px;
  margin: 0 10px;
  font-weight: 700;
`;
export const MenuContainer = styled.div`
  display: flex;
  align-items: center;
  background-color: #f0f0f0;
  padding: 30px;
`;

export const SubMenuContainer = styled.div``;
