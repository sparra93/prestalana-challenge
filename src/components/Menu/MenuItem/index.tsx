import type { ReactNode } from 'react';

import { Item } from './MenuItem.style';

interface MenuItemProps {
  children: ReactNode;
}

const MenuItem: React.FC<MenuItemProps> = ({ children }) => {
  return <Item>{children}</Item>;
};

export default MenuItem;
