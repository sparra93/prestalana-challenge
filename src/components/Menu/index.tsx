import Link from 'next/link';
import React from 'react';

import { MenuContainer, SubMenuContainer } from '@/components/Menu/Menu.style';
import MenuItem from '@/components/Menu/MenuItem';

const Menu: React.FC = () => {
  return (
    <MenuContainer>
      <MenuItem>Products</MenuItem>
      <SubMenuContainer>
        <Link href="/products/list" passHref data-testid="product-list-page">
          <MenuItem>List</MenuItem>
        </Link>
      </SubMenuContainer>
      <SubMenuContainer>
        <Link href="/cart" passHref data-testid="cart-page">
          <MenuItem>Cart</MenuItem>
        </Link>
      </SubMenuContainer>
    </MenuContainer>
  );
};

export default Menu;
