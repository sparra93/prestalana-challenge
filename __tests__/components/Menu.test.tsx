import { screen } from '@testing-library/react';
import React from 'react';

import Menu from '@/components/Menu';

describe('Menu Component', () => {
  it('should display Menu information correctly', () => {
    renderWithProviders(<Menu />);
    expect(screen.getByTestId('product-list-page')).toHaveAttribute(
      'href',
      '/products/list',
    );
  });

  it('should have a <Link/> that redirect to /products/list page', () => {
    renderWithProviders(<Menu />);

    const link = screen.getByTestId('product-list-page');
    expect(link).toBeInTheDocument();
  });

  it('should have a <Link/> that redirect to /cart page', () => {
    renderWithProviders(<Menu />);

    const link = screen.getByTestId('cart-page');
    expect(link).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProviders(<Menu />);

    expect(asFragment()).toMatchSnapshot();
  });
});
