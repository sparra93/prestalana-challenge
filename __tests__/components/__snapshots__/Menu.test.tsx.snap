// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-9101dc7-1 jBNtAt"
  >
    <div
      class="sc-9101dc7-0 elFxUS"
    >
      Opciones
    </div>
    <div
      class="sc-9101dc7-2"
    >
      <div
        class="sc-3a724431-0 iQabbx"
      >
        <a
          data-testid="product-list-page"
          href="/Products/List"
        >
          Products
        </a>
      </div>
    </div>
    <div
      class="sc-9101dc7-2"
    >
      <div
        class="sc-3a724431-0 iQabbx"
      >
        <a
          data-testid="cart-page"
          href="/Cart"
        >
          Cart
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
